### PR TITLE
#165: added form for eak qr-code generation

### PR DESCRIPTION
**Pull request by aiquita**
- eak form tile
- page consists of multiple form tiles
- Generate QR-codes button has not functionality yet!

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -1,6 +1,17 @@
 <component name="InspectionProjectProfileManager">
   <profile version="1.0">
     <option name="myName" value="Project Default" />
+    <inspection_tool class="CssUnknownProperty" enabled="true" level="WARNING" enabled_by_default="true">
+      <option name="myCustomPropertiesEnabled" value="true" />
+      <option name="myIgnoreVendorSpecificProperties" value="false" />
+      <option name="myCustomPropertiesList">
+        <value>
+          <list size="1">
+            <item index="0" class="java.lang.String" itemvalue="hegiht" />
```
Comment (maxammann) [inline .idea/inspectionProfiles/Project_Default.xml:10]: `hegiht` does not seem well spelled :D

File: administration/package.json [hunk 2]
```diff
@@ -19,6 +20,8 @@
     "react": "^17.0.1",
     "react-apollo": "^3.1.5",
     "react-dom": "^17.0.1",
+    "react-router": "^5.2.0",
```
Comment (maxammann) [inline administration/package.json:23]: I prefere the react-navigation router but probably it does not matter: https://www.npmjs.com/package/react-navigation

File: administration/src/components/generation/AddEakForm.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.add-eak-form-label {
```
Comment (maxammann) [inline administration/src/components/generation/AddEakForm.css:1]: Should we maybe use styled-components? What do you think @Schedulaar ?

File: administration/src/components/generation/EakGeneration.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+export class EakGeneration extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            cardCreationModels: [EakGeneration.createEmptyCard()]
+        };
+
+        this.addForm = this.addForm.bind(this);
```
Comment (maxammann) [inline administration/src/components/generation/EakGeneration.tsx:26]: The preferred way of writing components nowadays is using function components and react hooks.
That also avoids using constructors and binding funcitons to a context.

File: administration/src/components/generation/EakGeneration.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        this.reset = this.reset.bind(this);
+    }
+
+    componentDidMount(): void {
+        const firstFormColumn = document.getElementsByClassName("form-column")[0];
+        const addEakBpCard = document.getElementById("add-eak-form-card")
+        if (addEakBpCard != null) {
+            addEakBpCard.style.height = firstFormColumn.clientHeight - 50 + "px"
```
Comment (maxammann) [inline administration/src/components/generation/EakGeneration.tsx:34]: Why is that needed? Can you maybe set the height of the `form-column`s to a fixed height? That way you would not need to recalculate the height of elements

File: administration/src/components/generation/EakGeneration.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    }
+
+    private static createEmptyCard() {
```
Comment (maxammann) [inline administration/src/components/generation/EakGeneration.tsx:38]: I think the return type should be set here.

File: administration/src/models/CardCreationModel.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import {CardType} from "./CardType";
+
+export interface CardCreationModel {
```
Comment (maxammann) [inline administration/src/models/CardCreationModel.ts:3]: This model is only used in EakGeneation.ts. That means you can simply inline the interface there. I think right now there is no need for a separate `models` directory.

**Review by maxammann — APPROVED** [review]

File: administration/src/components/generation/EakGeneration.tsx (hunk 5 above)
Comment (aiquita) [reply administration/src/components/generation/EakGeneration.tsx:34]: I am not happy with this too, but as far as I can see there is no option to reach my goal otherwise. 

If I set form-column to a fixed height I fear that we maybe lose responsiveness of blueprintjs. As well, I do not want to investigate pixels by hand. 

CSS-Hacking does not help in this case, because not matter what I tried a edge case remains.

**Review by maxammann — APPROVED** [review]

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (Schedulaar) [reply .idea/inspectionProfiles/Project_Default.xml:10]: Still not

**Review by Schedulaar — COMMENTED** [review]
EakForm.css is empty and should be remoed

File: administration/src/components/generation/EakGeneration.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    const [cardCreationModels, setCardCreationModels] = useState([createEmptyCard()]);
+
+    /*componentDidMount(): void {
+        const firstFormColumn = document.getElementsByClassName("form-column")[0];
+        const addEakBpCard = document.getElementById("add-eak-form-card")
+        if (addEakBpCard != null) {
+            addEakBpCard.style.height = firstFormColumn.clientHeight - 50 + "px"
+        }
+    }*/
```
Comment (Schedulaar) [inline administration/src/components/generation/EakGeneration.tsx:30]: Remove commented code (or uncomment it if its neeed :) )

File: administration/src/components/generation/AddEakForm.css (hunk 3 above)
Comment (Schedulaar) [reply administration/src/components/generation/AddEakForm.css:1]: Yeah, would be nice, I think

File: administration/src/components/generation/EakGeneration.tsx (hunk 5 above)
Comment (Schedulaar) [reply administration/src/components/generation/EakGeneration.tsx:28]: If you really **need** to measure pixels here, use a react ref, pass the height via setState and prop to the related child element and define the style property on it.

Comment (Schedulaar) [reply administration/src/components/generation/EakGeneration.tsx:28]: What do you want to achieve by setting the height to `firstFormColumn.clientHeight - 50`?

**Comment by Schedulaar** [comment]
I added styled components in #180

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (maxammann) [reply .idea/inspectionProfiles/Project_Default.xml:10]: These changes should be reverted

**Review by maxammann — APPROVED** [review]

**Review by Schedulaar — APPROVED** [review]
